File: back-end/src/main/java/com/joaoalencar/projetoimpacta/domain/model/product/ProductImage.java
package com.joaoalencar.projetoimpacta.domain.model.product;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "product_images")
@Getter
@Setter
@NoArgsConstructor
public class ProductImage {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "file_name")
    private String fileName;

    @Column(name = "product_id")
    private int productId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id", insertable = false, updatable = false)
    private Product product;

    public ProductImage(String fileName, int productId) {
        this.fileName = fileName;
        this.productId = productId;
    }
}

File: api/Domain/Entities/ProductImage.cs
namespace api.Domain.Entities
{
    public class ProductImage
    {
        public long Id { get; set; }
        public string FileName { get; set; }
        public string ContentType { get; set; }
        public long ProductId { get; set; }

        public ProductImage()
        {
            
        }

        public ProductImage(string fileName, long productId, string contentType)
        {
            FileName = fileName;
            ProductId = productId;
            ContentType = contentType;
        }
    }
}